### PR TITLE
Adding support for borders to UILabel and UIButton (via CALayer and t…

### LINKED ISSUE #1884: [UIButton] Add Border Support
Add border support for UIButton

### PR DESCRIPTION
**Pull request by jaredhms**
…he CoreAnimation/UIKit composition layer).  Note that we still don't have border support on basic CALayers, because we haven't needed to add them yet, but the plumbing is in now place to add them when needed.

Fixes #1884.

### CODE REVIEW

File: Frameworks/UIKit/UITableViewCell.mm [hunk 1]
```diff
@@ -856,9 +856,9 @@ static void setInternalAccessoryColor(UITableViewCell* self) {
             }
         }
 
-        CGColorRef color =
-            static_cast<CGColorRef>([UIColor colorWithRed:contentColor.r green:contentColor.g blue:contentColor.b alpha:contentColor.a]);
-        [[self->_internalAccessoryView layer] _setContentColor:color];
+        // TODO: This used to call into a private/unimplemented CALayer function
+        // Not sure what we should be doing here.
+        UNIMPLEMENTED();
```
Comment (tadam-msft) [inline Frameworks/UIKit/UITableViewCell.mm]: What's the user impact?

Comment (jaredhms) [inline Frameworks/UIKit/UITableViewCell.mm]: Nothing - I deleted the empty private function we were calling.

File: samples/XAMLCatalog/XAMLCatalog/UIKitControls/UIButtonViewController.m [hunk 2]
```diff
@@ -78,13 +78,16 @@ - (UITableViewCell*)tableView:(UITableView*)tableView cellForRowAtIndexPath:(NSI
...
-        label.text = @"UIButton, no backgroundColor";
+        label.text = @"UIButton, no backgroundColor with green border";
 
         UIButton* button =
             [[UIButton alloc] initWithFrame:CGRectMake(_marginLeft, _marginTop + _labelHeight, _defaultWidth, _defaultHeight)];
         [button setTitle:@"Button" forState:UIControlStateNormal];
         [button setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
 
+        button.layer.borderColor = [[UIColor greenColor] CGColor];
+        button.layer.borderWidth = 1.0;
```
Comment (tadam-msft) [inline samples/XAMLCatalog/XAMLCatalog/UIKitControls/UIButtonViewController.m]: Should this update the new view controller Oliver is checking in--this is going to be deleted?

Comment (jaredhms) [inline samples/XAMLCatalog/XAMLCatalog/UIKitControls/UIButtonViewController.m]: He's not deleting this now, is he?  I figure we'll delete this when we add tests for the border.

File: Frameworks/UIKit.Xaml/Label.xaml.cpp [hunk 3]
```diff
@@ -53,12 +53,45 @@ Canvas^ Label::SublayerCanvas::get() {
     if (!_sublayerCanvas) {
         _sublayerCanvas = ref new Canvas();
         _sublayerCanvas->Name = "Sublayers";
-        Children->Append(_sublayerCanvas);
+
+        // If we have a border make sure the sublayer canvas is inserted before it
```
Comment (yiyang-msft) [inline Frameworks/UIKit.Xaml/Label.xaml.cpp]: is this to make sure that border's zindex is bigger than sublayer thus display on top?

Comment (jaredhms) [inline Frameworks/UIKit.Xaml/Label.xaml.cpp]: yes

**Comment by yiyang-msft** [comment]
:shipit:

**Pull request merged**